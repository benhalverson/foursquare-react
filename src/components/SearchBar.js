import React, { Component } from 'react';
class SearchBar extends Component {
	state = {
		term: this.props.term || '',
		location: this.props.location || 'San Jose, CA'
	};
	onInputChange = (e) => {
		this.setState({ term: e.target.value });
	};

	onInputChangeLocation = (e) => {
		this.setState({ location: e.target.value });
	};
	onFormSubmit = (e) => {
		e.preventDefault();
		this.props.onFormSubmit(this.state.term, this.state.location);
	};
	render() {
		return (
			<div className="ui segment">
				<form className="ui form" >
					<label>Video Search</label>
					<input type="text" className="field" value={this.state.term} onChange={this.onInputChange} />
					<input type="text" value={this.state.location} onChange={this.onInputChangeLocation} />
          <button onSubmit={this.onFormSubmit}>Search</button>
				</form>
			</div>
		);
	}
}

export default SearchBar;
