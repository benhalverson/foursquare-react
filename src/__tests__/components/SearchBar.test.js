import React from 'react';
import { shallow } from 'enzyme';
import '../../setupTests';
import SearchBar from '../../components/SearchBar';
const search = [
	{
		searchTerm: 'Dogs',
		location: 'San Jose, CA'
	}
];
test('should render SearchBar correctly', () => {
	const wrapper = shallow(<SearchBar />);
	expect(wrapper).toMatchSnapshot();
});

test('should set search on input change', () => {
	const value = 'Dogs';
	const wrapper = shallow(<SearchBar />);
	wrapper.find('FormControl').at(0).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('searchTerm')).toBe(value);
});

test('should set location on input change', () => {
	const value = 'San Jose, CA';
	const wrapper = shallow(<SearchBar />);
	wrapper.find('FormControl').at(0).simulate('change', {
		target: { value }
	});
	expect(wrapper.state('searchTerm')).toBe(value);
});

xtest('should call onSubmit prop for valid form submission', () => {
	const onSubmitSpy = jest.fn();
	const wrapper = shallow(<SearchBar searchTerm={search[0]} onFormSubmit={onSubmitSpy} />);
	wrapper.find('Form').simulate('submit', {
		preventDefault: () => {}
	});
	expect(wrapper.state('error')).toBe('');
	expect(onSubmitSpy).toHaveBeenLastCalledWith({
		searchTerm: search[0].searchTerm,
		location: search[0].location
	});
});
