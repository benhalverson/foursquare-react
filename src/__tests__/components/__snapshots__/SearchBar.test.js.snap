// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchBar correctly 1`] = `ShallowWrapper {}`;
